From 8cd6d527d017bbd35552a2ce0042c056c34a7ee2 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 3 Jan 2018 18:22:03 +0800
Subject: [PATCH 3157/5242] MLK-17334: imx8: pm-domains: fix array overflow
 issue

commit  f71a053a80fcaf6b9fe749ada868d867aa1f8cdc from
https://source.codeaurora.org/external/imx/linux-imx.git

When the resource id is larger than 512, the wakeup_rsrc_id array
will overflow, then the resource may always power on.
So align the IRQ with resource number to fix the issue.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index 2361150..bef3680 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -42,7 +42,7 @@
 };
 static sc_rsrc_t rsrc_debug_console;
 
-#define IMX8_WU_MAX_IRQS	512
+#define IMX8_WU_MAX_IRQS	(((SC_R_LAST + 31) / 32 ) * 32 )
 static sc_rsrc_t irq2rsrc[IMX8_WU_MAX_IRQS];
 static sc_rsrc_t wakeup_rsrc_id[IMX8_WU_MAX_IRQS / 32];
 static DEFINE_SPINLOCK(imx8_wu_lock);
-- 
1.7.9.5

